File: aemet/SergioDeIscar-aemet-java/SergioDeIscar-aemet-java/src/main/java/org/example/dto/DuplasDto.java
package org.example.dto;

import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.io.Serializable;
import java.util.List;

@Root(name = "duplas")
public class DuplasDto implements Serializable {
    public DuplasDto(List<DuplaDto> duplas) {
        this.duplas = duplas;
    }
    public DuplasDto(){
        // Constructor vacio para SimpleXML
    }
    @ElementList(name = "dupla", inline = true)
    private List<DuplaDto> duplas;

    //Getters
    public List<DuplaDto> getDuplas() {
        return duplas;
    }

}

File: aemet/SergioDeIscar-aemet-kotlin/SergioDeIscar-aemet-kotlin/src/main/kotlin/dto/DuplasDto.kt
package dto

import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

@Root(name = "duplas")
data class DuplasDto(
    @field:ElementList(name = "dupla", inline = true)
    @param:ElementList(name = "dupla", inline = true)
    val duplas: List<DuplaDto>
)
